frappe/model/document.py

```diff
@@ -1143,6 +1143,23 @@ def add_seen(self, user=None):
 				self.db_set('_seen', json.dumps(_seen), update_modified=False)
 				frappe.local.flags.commit = True
 
+	def add_viewed(self, user=None):
+		'''add log to communication when a user viewes a document'''
+		if not user:
+			user = frappe.session.user
+
+		if self.meta.track_views:
+			frappe.get_doc({
```

Comment:
not sure if communication is the right table for this... shall we create another table to partition the info?

Reply:
The structure of the new doctype should be same as communication doctype?

Reply:
No can be much smaller! How about `View Log` ?